### PR TITLE
aws-janitor: normalize generated arn ids for resources

### PR DESCRIPTION
**Pull request by cpanato**
Normalize the ARN for resources that do not have.

/assign @ixdy 

please let me know if that makes sense or what you think, thanks for your review in advance :)

Fixes: https://github.com/kubernetes-sigs/boskos/issues/55

### CODE REVIEW

File: aws-janitor/resources/asg.go [hunk 1]
```diff
@@ -103,12 +106,14 @@ func (AutoScalingGroups) ListAll(opts Options) (*Set, error) {
...
-	ID   string
-	Name string
+	Account string
+	Region  string
+	ID      string
+	Name    string
 }
 
 func (asg autoScalingGroup) ARN() string {
-	return asg.ID
```
Comment (ixdy) [inline aws-janitor/resources/asg.go:111]: I think this one is already using the ARN (see L41), it's just named improperly in the struct.

File: aws-janitor/resources/efs.go [hunk 2]
```diff
@@ -123,12 +126,13 @@ func (ElasticFileSystems) ListAll(opts Options) (*Set, error) {
...
 
 type elasticFileSystem struct {
-	ID string
+	Account string
+	Region  string
+	ID      string
 }
 
 func (efs elasticFileSystem) ARN() string {
-	// The current client library does not provide ARNs for file systems.
```
Comment (ixdy) [inline aws-janitor/resources/efs.go:130]: I think this comment is actually false? (or maybe it was added at some point?) 

we should be able to use the ARN provided in the result: https://docs.aws.amazon.com/sdk-for-go/api/service/efs/#FileSystemDescription

File: aws-janitor/resources/cloud_formation_stacks.go [hunk 3]
```diff
@@ -87,12 +96,14 @@ func (CloudFormationStacks) ListAll(opts Options) (*Set, error) {
...
-	name string
+	account string
+	region  string
+	id      string
+	name    string
 }
 
 func (p cloudFormationStack) ARN() string {
-	return p.id
+	return fmt.Sprintf("arn:aws:cloud-formation:%s:%s:stack/%s", p.region, p.account, p.id)
```
Comment (ixdy) [inline aws-janitor/resources/cloud_formation_stacks.go:106]: nit: it seems like most of the real ARNs don't use dashes in the service name (i.e. make this `cloudformation`, not `cloud-formation`)

File: aws-janitor/resources/elb.go [hunk 4]
```diff
@@ -96,7 +97,7 @@ type classicLoadBalancer struct {
 }
 
 func (lb classicLoadBalancer) ARN() string {
-	return "fakearn:elb:" + lb.region + ":" + lb.account + ":" + lb.dnsName
+	return fmt.Sprintf("arn:aws:ec2:%s:%s:classic-elb/%s", lb.region, lb.account, lb.dnsName)
```
Comment (ixdy) [inline aws-janitor/resources/elb.go:100]: nit: service probably should be something like `elb`, not `ec2`

**Review by ixdy — COMMENTED** [review]
thanks! just a few comments.

File: aws-janitor/resources/efs.go (hunk 2 above)
Comment (cpanato) [reply aws-janitor/resources/efs.go:130]: to use that, i've bumped the `github.com/aws/aws-sdk-go` dependency

File: aws-janitor/resources/asg.go (hunk 1 above)
Comment (cpanato) [reply aws-janitor/resources/asg.go:111]: true! thanks for pointing out

thanks for your review @ixdy 
PTAL

**Review by ixdy — APPROVED** [review]
Sorry for the long review cycle on this. Thanks for the PR, I do appreciate it!

**Pull request merged**